src/plugins/data/server/index_patterns/fetcher/lib/opensearch_api.ts

```diff
@@ -84,11 +93,20 @@ export async function callIndexAliasApi(
  *  @return {Promise<FieldCapsResponse>}
  */
 export async function callFieldCapsApi(
-  callCluster: LegacyAPICaller,
+  callCluster: LegacyAPICaller | OpenSearchClient,
   indices: string[] | string,
   fieldCapsOptions: { allowNoIndices: boolean } = { allowNoIndices: false }
 ) {
   try {
+    if ('transport' in callCluster) {
```

Comment:
does this `transport` means multiple datasource remote connection?

Reply:
No, it's just the typescript way to do type guard. in this case, the object which contains `transport` field should be a OpenSearchClient

```
export type OpenSearchClient = Omit<
  OpenSearchDashboardsClient,
  'connectionPool' | 'transport' | 'serializer' | 'extend' | 'child' | 'close'
> & {
  transport: {
    request(
      params: TransportRequestParams,
      options?: TransportRequestOptions
    ): TransportRequestPromise<ApiResponse>;
  };
};
```

https://stackoverflow.com/questions/14425568/interface-type-check-with-typescript

Reply:
And this approach should be replaced once we support legacy client with new data client.